# EDA con Enfoque en CAC y LTV

Este cuaderno se centra en **preguntas de negocio** que generan valor al entender la relación entre:
- **CAC**: Costo de Adquisición de Cliente.
- **LTV**: Valor de Vida del Cliente.

Adicionalmente, explora aspectos clave como **suscripción**, **descuentos**, **categoría** del producto, **frecuencia de compras**, **método de pago**, etc. Con estas preguntas podrás tomar **decisiones estratégicas** sobre tu modelo de negocio, campañas de marketing y retención de clientes.

## Contenido:
1. **Importar librerías y cargar datos**
2. **Inspección inicial** (dimensiones, tipos de datos, nulos)
3. **Limpieza básica** (si es necesaria)
4. **Creación de columna** ratio `LTV/CAC` (rentabilidad por cliente)
5. **Preguntas de negocio** (groupby, filtros, etc.)
6. **Conclusiones**

In [1]:
!wget !wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv

--2025-03-12 21:46:22--  http://!wget/
Resolving !wget (!wget)... failed: Name or service not known.
wget: unable to resolve host address ‘!wget’
--2025-03-12 21:46:22--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv
Resolving github.com (github.com)... 140.82.116.3
Connecting to github.com (github.com)|140.82.116.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv [following]
--2025-03-12 21:46:23--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/shopping_trends_LTV_CAC.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HT

## 1. Importar librerías y cargar datos
Asegúrate de tener tu archivo CSV con las columnas mencionadas: `CAC`, `LTV`, `Purchase Amount (USD)`, etc. Ajusta la ruta según tu archivo real.

In [2]:
import pandas as pd
import numpy as np


# Cargar el dataset
df = pd.read_csv("shopping_trends_LTV_CAC.csv")


# Mostramos las primeras filas para verificar
df.head()

Customer ID  Age Gender Item Purchased  Category  Purchase Amount (USD)  \
0            1   55   Male         Blouse  Clothing                     53   
1            2   19   Male        Sweater  Clothing                     64   
2            3   50   Male          Jeans  Clothing                     73   
3            4   21   Male        Sandals  Footwear                     90   
4            5   45   Male         Blouse  Clothing                     49   

        Location Size      Color  Season  ...  Shipping Type Discount Applied  \
0       Kentucky    L       Gray  Winter  ...        Express              Yes   
1          Maine    L     Maroon  Winter  ...        Express              Yes   
2  Massachusetts    S     Maroon  Spring  ...  Free Shipping              Yes   
3   Rhode Island    M     Maroon  Spring  ...   Next Day Air              Yes   
4         Oregon    M  Turquoise  Spring  ...  Free Shipping              Yes   

  Promo Code Used Previous Purchases Preferred Payment Method  \
0             Yes                 14                    Venmo   
1             Yes                  2                     Cash   
2             Yes                 23              Credit Card   
3             Yes                 49                   PayPal   
4             Yes                 31                   PayPal   

  Frequency of Purchases    CAC   LTV Rentabilidad  LTV_CAC_Ratio  
0            Fortnightly  112.0  1512    16.434783      13.500000  
1            Fortnightly   95.0  1497    19.960000      15.757895  
2                 Weekly   32.0  1137    94.750000      35.531250  
3                 Weekly    NaN  1914    26.957746      26.957746  
4               Annually   47.0  1300    48.148148      27.659574  

[5 rows x 23 columns]

## 2. Inspección inicial
1. **Dimensiones (filas x columnas)**
2. **Info general**: tipos de dato, nulos, etc.
3. **Valores nulos** por columna.

Si existen valores nulos en columnas importantes (por ejemplo, `CAC`, `LTV`, `Customer ID`), evalúa si los eliminarás o imputarás.

In [3]:
#Columnas

df.shape

(3900, 23)

In [4]:
#Informacion general
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3900 entries, 0 to 3899
Data columns (total 23 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Customer ID               3900 non-null   int64  
 1   Age                       3900 non-null   int64  
 2   Gender                    3900 non-null   object 
 3   Item Purchased            3900 non-null   object 
 4   Category                  3900 non-null   object 
 5   Purchase Amount (USD)     3900 non-null   int64  
 6   Location                  3900 non-null   object 
 7   Size                      3900 non-null   object 
 8   Color                     3900 non-null   object 
 9   Season                    3900 non-null   object 
 10  Review Rating             3900 non-null   float64
 11  Subscription Status       3900 non-null   object 
 12  Payment Method            3900 non-null   object 
 13  Shipping Type             3900 non-null   object 
 14  Discount

In [5]:
df.isnull().sum()

Customer ID                   0
Age                           0
Gender                        0
Item Purchased                0
Category                      0
Purchase Amount (USD)         0
Location                      0
Size                          0
Color                         0
Season                        0
Review Rating                 0
Subscription Status           0
Payment Method                0
Shipping Type                 0
Discount Applied              0
Promo Code Used               0
Previous Purchases            0
Preferred Payment Method      0
Frequency of Purchases        0
CAC                         599
LTV                           0
Rentabilidad                  0
LTV_CAC_Ratio                 0
dtype: int64

In [ ]:
#Descripción
df.describe()

Customer ID          Age  Purchase Amount (USD)  Review Rating  \
count  3900.000000  3900.000000            3900.000000    3900.000000   
mean   1950.500000    44.068462              59.764359       3.749949   
std    1125.977353    15.207589              23.685392       0.716223   
min       1.000000    18.000000              20.000000       2.500000   
25%     975.750000    31.000000              39.000000       3.100000   
50%    1950.500000    44.000000              60.000000       3.700000   
75%    2925.250000    57.000000              81.000000       4.400000   
max    3900.000000    70.000000             100.000000       5.000000   

       Previous Purchases          CAC          LTV  Rentabilidad  \
count         3900.000000  3301.000000  3900.000000   3900.000000   
mean            25.351538    66.292336  1108.944872     28.330611   
std             14.447125    27.712380   527.645495     27.627277   
min              1.000000    10.000000   200.000000      2.070707   
25%             13.000000    44.000000   657.000000     11.665094   
50%             25.000000    67.000000  1107.000000     19.784776   
75%             38.000000    88.000000  1563.000000     33.307253   
max             50.000000   120.000000  2410.000000    215.800000   

       LTV_CAC_Ratio  
count    3900.000000  
mean       23.301258  
std        22.705725  
min         1.722689  
25%        10.035880  
50%        17.062261  
75%        27.629467  
max       215.800000

## 3. Limpieza básica
Por ejemplo, si detectas `CAC <= 0` o `LTV <= 0` y consideras que no tienen sentido en tu negocio, podrías filtrar esos registros. Ajusta según tu criterio.

In [ ]:
df= df.dropna(subset=['CAC'])

In [ ]:
df.isnull().sum()

Customer ID                 0
Age                         0
Gender                      0
Item Purchased              0
Category                    0
Purchase Amount (USD)       0
Location                    0
Size                        0
Color                       0
Season                      0
Review Rating               0
Subscription Status         0
Payment Method              0
Shipping Type               0
Discount Applied            0
Promo Code Used             0
Previous Purchases          0
Preferred Payment Method    0
Frequency of Purchases      0
CAC                         0
LTV                         0
Rentabilidad                0
LTV_CAC_Ratio               0
dtype: int64

In [ ]:
df[df['LTV']==""]

Empty DataFrame
Columns: [Customer ID, Age, Gender, Item Purchased, Category, Purchase Amount (USD), Location, Size, Color, Season, Review Rating, Subscription Status, Payment Method, Shipping Type, Discount Applied, Promo Code Used, Previous Purchases, Preferred Payment Method, Frequency of Purchases, CAC, LTV, Rentabilidad, LTV_CAC_Ratio]
Index: []

[0 rows x 23 columns]

In [ ]:
# Eliminamos filas donde CAC menor o igual a 0
df[df['CAC']==""]

Empty DataFrame
Columns: [Customer ID, Age, Gender, Item Purchased, Category, Purchase Amount (USD), Location, Size, Color, Season, Review Rating, Subscription Status, Payment Method, Shipping Type, Discount Applied, Promo Code Used, Previous Purchases, Preferred Payment Method, Frequency of Purchases, CAC, LTV, Rentabilidad, LTV_CAC_Ratio]
Index: []

[0 rows x 23 columns]

In [ ]:
df[df["LTV"]<=0]

Empty DataFrame
Columns: [Customer ID, Age, Gender, Item Purchased, Category, Purchase Amount (USD), Location, Size, Color, Season, Review Rating, Subscription Status, Payment Method, Shipping Type, Discount Applied, Promo Code Used, Previous Purchases, Preferred Payment Method, Frequency of Purchases, CAC, LTV, Rentabilidad, LTV_CAC_Ratio]
Index: []

[0 rows x 23 columns]

## 4. Creación de columna: Ratio LTV/CAC
Es una métrica clave de la **rentabilidad** del cliente. Un valor > 1 indica que, en teoría, el cliente retorna más de lo que cuesta adquirirlo.

In [ ]:
df['Ratio'] = df['LTV'] / df['CAC']

In [ ]:
df.head()

Customer ID  Age Gender Item Purchased  Category  Purchase Amount (USD)  \
0            1   55   Male         Blouse  Clothing                     53   
1            2   19   Male        Sweater  Clothing                     64   
2            3   50   Male          Jeans  Clothing                     73   
4            5   45   Male         Blouse  Clothing                     49   
6            7   63   Male          Shirt  Clothing                     85   

        Location Size      Color  Season  ...  Discount Applied  \
0       Kentucky    L       Gray  Winter  ...               Yes   
1          Maine    L     Maroon  Winter  ...               Yes   
2  Massachusetts    S     Maroon  Spring  ...               Yes   
4         Oregon    M  Turquoise  Spring  ...               Yes   
6        Montana    M       Gray    Fall  ...               Yes   

  Promo Code Used Previous Purchases Preferred Payment Method  \
0             Yes                 14                    Venmo   
1             Yes                  2                     Cash   
2             Yes                 23              Credit Card   
4             Yes                 31                   PayPal   
6             Yes                 49                     Cash   

  Frequency of Purchases    CAC   LTV Rentabilidad LTV_CAC_Ratio      Ratio  
0            Fortnightly  112.0  1512    16.434783     13.500000  13.500000  
1            Fortnightly   95.0  1497    19.960000     15.757895  15.757895  
2                 Weekly   32.0  1137    94.750000     35.531250  35.531250  
4               Annually   47.0  1300    48.148148     27.659574  27.659574  
6              Quarterly   99.0   213     2.696203      2.151515   2.151515  

[5 rows x 24 columns]

## 5. Preguntas de Negocio
En vez de usar gráficas o correlaciones, nos centramos en **preguntas que generen valor** al análisis. Cada una se resuelve con **groupby, filtros o estadísticas** simples.

1. **¿Cuántos clientes únicos hay y cuál es la edad promedio?**
2. **¿Cuál es el CAC promedio y el LTV promedio en todo el dataset?**
3. **¿Cuál es la ratio LTV/CAC promedio?**
4. **¿Qué categoría tiene el mayor LTV en promedio?**
5. **¿En cuál categoría se justifica más el CAC?** (categoría con mayor ratio LTV/CAC)
6. **¿Los clientes suscritos tienen LTV mayor que los no suscritos?**
7. **¿Cuál es el método de pago más frecuente y su LTV promedio?**
8. **¿Quiénes usan descuentos con mayor frecuencia?** (Por ejemplo, ¿cierto género, cierto rango de edad?)
9. **¿Cuántos clientes aprovechan códigos promocionales (Promo Code Used)?** y su ratio LTV/CAC
10. **¿Cuál es la frecuencia de compra más común (Frequency of Purchases)?** y su LTV asociado.
11. **¿Cuál es la ubicación (Location) con mayor ratio LTV/CAC?**
12. **Entre los clientes con `Previous Purchases` > X, cuál es el gasto promedio y LTV promedio?**


### 5.1 ¿Cuántos clientes únicos hay y cuál es la edad promedio?

RTA: 3301 son el total de clientes únicos y tienen un promedio de 44 años.

In [ ]:
df['Customer ID'].nunique()

3301

In [ ]:
df['Age'].mean()

44.00060587700697

### 5.2 ¿Cuál es el CAC promedio y el LTV promedio en todo el dataset?

RTA: El CAC promedio es de 66.29 y la LTV promedio es 1110.69.

In [ ]:
df['CAC'].mean()

66.29233565586186

In [ ]:
df['LTV'].mean()

1110.6934262344744

### 5.3 ¿Cuál es la ratio LTV/CAC promedio?

RTA: El Ratio promedio es 22.44.

In [ ]:
df['Ratio'].mean()

22.446982949947316

### 5.4 ¿Qué categoría tiene el mayor LTV en promedio?

RTA: Accesories es la categoría con mayor LTV promedio (1145.99).

In [ ]:
df.groupby('Category')['LTV'].mean()

Category
Accessories    1145.992742
Clothing       1095.688543
Outerwear      1056.040123
Name: LTV, dtype: float64

### 5.5 ¿En cuál categoría se justifica más el CAC? (mayor ratio LTV/CAC)

RTA: Es Accesories la categoría con ratio (29.28).

In [ ]:
df.groupby('Category')['Ratio'].mean()

Category
Accessories    29.281313
Clothing       16.845818
Outerwear      26.319371
Name: Ratio, dtype: float64

### 5.6 ¿Los clientes suscritos tienen LTV mayor que los no suscritos?

RTA: Los clientes que tiene mayor LTV son los que no están suscritos.

In [ ]:
df.groupby('Subscription Status')['LTV'].mean()

Subscription Status
No     1115.798264
Yes    1096.692744
Name: LTV, dtype: float64

### 5.7 ¿Cuál es el método de pago más frecuente y su LTV promedio?

RTA: El método de pago más frecuente es Credit Card y tiene un LTV de 1099.99.

In [ ]:
df['Payment Method'].value_counts()

Payment Method
Credit Card      612
Venmo            545
Cash             543
Debit Card       542
Bank Transfer    534
PayPal           525
Name: count, dtype: int64

In [ ]:
df.groupby('Payment Method')['LTV'].mean()

Payment Method
Bank Transfer    1135.986891
Cash             1162.265193
Credit Card      1099.990196
Debit Card       1098.097786
PayPal           1086.798095
Venmo            1082.091743
Name: LTV, dtype: float64

### 5.8 ¿Quiénes usan descuentos con mayor frecuencia?
Podemos ver si hay algún grupo (por género, rango de edad, etc.) que destaca. A modo de ejemplo, segmentaremos por género:

RTA: Male utiliza más los descuentos.

In [ ]:
df.groupby('Gender')['Discount Applied'].count()

Gender
Female    1049
Male      2252
Name: Discount Applied, dtype: int64

Podrías repetir el ejercicio segmentando por **Age** (definiendo rangos) o por **Location**.

### 5.9 ¿Cuántos clientes aprovechan códigos promocionales y cuál es su ratio LTV/CAC?
El uso de promos puede modificar la rentabilidad.

RTA: 1418 clientes utilizan código promocional, los que utilizan código tienen un ratio de 22.70.

In [ ]:
df['Promo Code Used'].value_counts()

Promo Code Used
No     1883
Yes    1418
Name: count, dtype: int64

In [ ]:
df.groupby('Promo Code Used')['Ratio'].mean()

Promo Code Used
No     22.252655
Yes    22.705036
Name: Ratio, dtype: float64

### 5.10 ¿Cuál es la frecuencia de compra más común (Frequency of Purchases) y su LTV?
Esto ayuda a ver si, por ejemplo, los clientes que compran semanalmente tienen un LTV más alto.

RTA: Every 3 Months es la frecuencia de compra y tiene un LTV de 1088.77.

In [ ]:
df['Frequency of Purchases'].value_counts()

Frequency of Purchases
Every 3 Months    492
Annually          485
Quarterly         480
Monthly           478
Bi-Weekly         467
Weekly            453
Fortnightly       446
Name: count, dtype: int64

In [ ]:
df.groupby('Frequency of Purchases')['LTV'].mean()

Frequency of Purchases
Annually          1095.987629
Bi-Weekly         1181.297645
Every 3 Months    1088.776423
Fortnightly       1112.547085
Monthly           1101.855649
Quarterly         1084.364583
Weekly            1112.854305
Name: LTV, dtype: float64

### 5.11 ¿Cuál es la ubicación (Location) con mayor ratio LTV/CAC?

RTA: New York ubicación con mayor Ratio.

In [ ]:
df.groupby('Location')['Ratio'].mean()

Location
Alabama           21.481423
Alaska            21.457456
Arizona           23.806429
Arkansas          20.151509
California        27.714002
Colorado          21.680277
Connecticut       21.473588
Delaware          20.915415
Florida           23.776268
Georgia           19.392744
Hawaii            27.288639
Idaho             18.601725
Illinois          22.232519
Indiana           23.329325
Iowa              21.709956
Kansas            21.613887
Kentucky          30.392803
Louisiana         23.645367
Maine             18.875671
Maryland          24.957436
Massachusetts     19.090060
Michigan          22.123296
Minnesota         19.024386
Mississippi       23.258348
Missouri          18.782766
Montana           23.426980
Nebraska          19.903884
Nevada            17.566468
New Hampshire     23.870782
New Jersey        24.251955
New Mexico        22.680627
New York          32.785849
North Carolina    22.136964
North Dakota      26.591688
Ohio              25.319319
Oklahoma          22.231900
Oregon            21.181867
Pennsylvania      22.421718
Rhode Island      25.084850
South Carolina    21.768980
South Dakota      30.023524
Tennessee         17.331910
Texas             18.933904
Utah              19.700658
Vermont           21.189444
Virginia          22.030869
Washington        21.299870
West Virginia     20.527099
Wisconsin         20.351115
Wyoming           21.590420
Name: Ratio, dtype: float64

### 5.12 Entre los clientes con `Previous Purchases` > X, ¿cuál es el gasto promedio y el LTV promedio?
Define un umbral X según tu contexto. Por ejemplo, X = 20:

RTA = Los clientes con un Previous Purchases mayor a 20, tienen promedio de gasto es 66.10 y el LTV 1121.81.

In [ ]:
dfmayora20 =df[df['Previous Purchases'] > 20]

In [ ]:
dfmayora20['CAC'].mean()

66.1076688674454

In [ ]:
dfmayora20['LTV'].mean()

1121.810563737938

## 6. Conclusiones

En este Notebook, hemos abordado un **EDA orientado a negocio** en el que:
1. **Identificamos** la base total de clientes y su edad promedio.
2. **Calculamos** el CAC y LTV promedio, y la ratio LTV/CAC global.
3. **Segregamos** por categorías, suscripción, uso de descuentos, métodos de pago, etc.
4. **Resolvimos** preguntas de negocio que ayudan a:
   - **Identificar** cuáles categorías o ubicaciones ofrecen mejor rentabilidad (ratio LTV/CAC).
   - **Entender** si la suscripción eleva el LTV.
   - **Explorar** la efectividad de promos y descuentos.
   - **Diferenciar** frecuencias de compra y ver su impacto en LTV.
   - **Focalizar** inversiones de marketing en los segmentos más prometedores.

Este tipo de **análisis** proporciona las bases para **estrategias** de retención, adquisición y segmentación de clientes. ¡Espero que te sea útil para **generar valor** en tu negocio!